lib/compliance/flows/id-card-data.js

```diff
@@ -10,34 +10,42 @@ const idDataMachine = xstate.Machine({
   initial: 'idle',
   strict: true,
   states: {
-    idle: {on: {START: 'scanId'}},
+    idle: { on: { START: 'scanId' } },
     scanId: {
       onEntry: ['timeoutToScannerCancel', 'transitionScreen', 'scanPDF'],
       on: {
-        SCANNED: {'authorizing': {actions: ['transitionScreen', 'authorizeIdCardData']}},
-        SCAN_ERROR: {'idScanFailed': {actions: ['timeoutToFail', 'transitionScreen']}},
+        SCANNED: { 'authorizing': { actions: ['transitionScreen', 'authorizeIdCardData'] } },
+        SCAN_ERROR: { 'idScanFailed': { actions: ['timeoutToFail', 'transitionScreen'] } },
         FAIL: 'failure'
       }
     },
     authorizing: {
       on: {
         AUTHORIZED: 'success',
-        BLOCKED_ID: {'idVerificationFailed': {actions: ['timeoutToFail', 'transitionScreen']}}
+        BLOCKED_ID: { 'idVerificationFailed': { actions: ['timeoutToFail', 'transitionScreen'] } }
       }
     },
-    idScanFailed: {on: {FAIL: 'failure'}},
-    idVerificationFailed: {on: {FAIL: 'failure'}},
+    idScanFailed: { on: { FAIL: 'failure' } },
+    idVerificationFailed: { on: { FAIL: 'failure' } },
     failure: { onEntry: ['failure'] },
     success: { onEntry: ['success'] }
   }
 })
 
 let currentStateValue
 let data
+let idCardDataStills = []
 
 function getData () { return data }
 function setData (value) { data = value }
 
+function addIdDataPhoto (photo) {
+  idCardDataStills.push(photo)
+  if (_.size(idCardDataStills) > 5) idCardDataStills = idCardDataStills.slice(5)
```

Comment:
If you want to take the last 5 members of `idCardDataStills`, I think you can just do

```suggestion
  if (_.size(idCardDataStills) > 5) idCardDataStills = _.takeRight(5, idCardDataStills)
```